File: Data Structures & Algorithms/online-stock-span/submission-0.cpp
class StockSpanner {
public:
    vector<pair<int, int>> st;
    StockSpanner() = default;
    
    int next(int price) {
        int sp = 1;
        while (true) {
            if (st.size() == 0) {
                st.push_back(make_pair(price, sp));
                return sp;
            }
            if (price >= st.back().first) {
                sp += st.back().second;
                st.pop_back();
            } else {
                break;
            }
        }
        st.push_back(make_pair(price, sp));
        return sp;
    }
};

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner* obj = new StockSpanner();
 * int param_1 = obj->next(price);
 */

File: Data Structures & Algorithms/maximum-frequency-stack/submission-0.py
class FreqStack:

    def __init__(self):
        self.heap = []
        self.freq = {}

    def push(self, val: int) -> None:
            fr = self.freq.get(val, 0) + 1
            self.freq[val] = fr
            heapq.heappush(self.heap, (-fr, -len(self.heap), val))

    def pop(self) -> int:
        r = heapq.heappop(self.heap)
        self.freq[r[2]] -= 1
        return r[2]


# Your FreqStack object will be instantiated and called as such:
# obj = FreqStack()
# obj.push(val)
# param_2 = obj.pop()